src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -3065,20 +3065,22 @@ module Mutations = struct
         Some (Mina_commands.reset_trust_status coda ip_address) )
 
   let send_user_command coda user_command_input =
-    match
-      Mina_commands.setup_and_submit_user_command coda user_command_input
-    with
-    | `Active f -> (
-        match%map f with
-        | Ok user_command ->
-            Ok
-              { Types.UserCommand.With_status.data = user_command
-              ; status = Unknown
-              }
-        | Error e ->
-            Error ("Couldn't send user_command: " ^ Error.to_string_hum e) )
-    | `Bootstrapping ->
-        return (Error "Daemon is bootstrapping")
+    match Mina_compile_config.slot_tx_end with
```

Comment:
In one of the first iterations of the corresponding RFC, it was discussed changing the transaction pool logic, but it was decided to keep it unchanged.

As for rejecting txns at the CLI level, it was suggested to do it at the GraphQL layer. I checked the code, and calling the CLI to send payments uses GraphQL, so, inherently, txns are rejected when using CLI.

On an initial implementation that changed the txn pool, txns were being rejected, either if they came from CLI/GraphQL or the gossip network. In this current implementation, we have no mechanism to deal with txns coming from the gossip network. Regardless, rejecting txns is not necessary to produce empty blocks, it was added simply as an improvement to UX.

Please let me know if you need me to address this in another way.

Reply:
The CLI doesn't always use graphql. For instance, sending batches of payments does not use graphql, it uses the rpc interface https://github.com/MinaProtocol/mina/blob/develop/src/app/cli/src/init/client.ml#L433.

I would still urge you to move this error into the transaction pool so that we have full coverage of all the places where a user can submit commands.